reframe/core/schedulers/slurm.py

```diff
@@ -68,6 +69,17 @@ class SlurmJob(sched.Job):
     # standard job state polling using sacct.
     SACCT_SQUEUE_RATIO = 10
 
+    @staticmethod
+    def _get_nodes_from_description(descriptions):
```

Comment:
I would make this a free function named `create_nodes()` defined just before the `SlurmNode` (not at the beginning of the file).